validator/db/kv/attestation_history.go

```diff
@@ -81,20 +110,98 @@ func (store *Store) SaveAttestationHistoryForPubKeys(
 				return err
 			}
 		}
+		for pubKey, min := range minByPubKeys {
+			if err := bucket.Put(attHist.GetMinSourceKey(pubKey), bytesutil.Uint64ToBytesBigEndian(min.Source)); err != nil {
+				return err
+			}
+			if err := bucket.Put(attHist.GetMinTargetKey(pubKey), bytesutil.Uint64ToBytesBigEndian(min.Target)); err != nil {
+				return err
+			}
+		}
 		return nil
 	})
 	return err
 }
 
 // SaveAttestationHistoryForPubKey saves the attestation history for the requested validator public key.
 func (store *Store) SaveAttestationHistoryForPubKey(
-	ctx context.Context, pubKey [48]byte, history attHist.History,
+	ctx context.Context,
+	pubKey [48]byte,
+	history attHist.History,
 ) error {
 	ctx, span := trace.StartSpan(ctx, "Validator.SaveAttestationHistoryForPubKey")
 	defer span.End()
 	err := store.update(func(tx *bolt.Tx) error {
 		bucket := tx.Bucket(newHistoricAttestationsBucket)
-		return bucket.Put(pubKey[:], history)
+		if err := bucket.Put(pubKey[:], history); err != nil {
+			return err
+		}
+
+		return nil
 	})
 	return err
 }
+
+// SaveMinAttestation saves min attestation values if they are lower from the ones that are currently set in db.
+func (store *Store) SaveMinAttestation(ctx context.Context, pubKey [48]byte, minAtt attHist.MinAttestation) error {
```

Comment:
The name MinAttestation isn't very informative. We need to rename this to something more explicit that is clear regardless of someone's background in slashing protection. I propose something like `EarliestSignedAttestationForPubKey` and make public functions follow that convention. `EarliestSignedAttestation` is more readable